test/hotspot/jtreg/compiler/ciReplay/SABase.java

```diff
@@ -96,19 +110,42 @@ public void testAction() {
         File replay = new File(REPLAY_FILE_NAME);
         Asserts.assertTrue(replay.exists() && replay.isFile() && replay.length() > 0,
                 "Replay data wasn't generated by SA");
+        // other than comment lines, content of 2 files should be identical
         try {
-            FileInputStream rep = new FileInputStream(replay);
-            FileInputStream repCopy = new FileInputStream(REPLAY_FILE_COPY);
-            byte repBuffer[] = new byte[512];
-            byte repCopyBuffer[] = new byte[512];
-            boolean filesNotEqual = false;
-            while(rep.available() > 0 && !filesNotEqual) {
-                int count = rep.read(repBuffer);
-                int count2 = repCopy.read(repCopyBuffer);
-                filesNotEqual = count != count2 || Arrays.equals(repBuffer, repCopyBuffer);
+            BufferedReader rep = new BufferedReader(new FileReader(replay));
+            BufferedReader repCopy = new BufferedReader(new FileReader(REPLAY_FILE_COPY));
+            boolean failure = false;
+            while (true) {
+                String l1;
+                while ((l1 = rep.readLine()) != null) {
+                    if (!l1.startsWith("#")) {
+                        break;
+                    }
+                }
+                String l2;
+                while ((l2 = repCopy.readLine()) != null) {
+                    if (!l2.startsWith("#")) {
+                        break;
+                    }
+                }
+                if (l1 == null || l2 == null) {
+                    if (l1 != null || l2 != null) {
+                        System.out.println("Warning: replay files are not equal");
+                        System.out.println("1: " + l1);
+                        System.out.println("2: " + l2);
+                        failure = true;
+                    }
+                    break;
+                }
+                if (!l1.equals(l2)) {
+                    System.out.println("Warning: replay files are not equal");
+                    System.out.println("1: " + l1);
+                    System.out.println("2: " + l2);
+                    failure = true;
```

Comment:
Don't we want to break out of the loop at this point?

Reply:
I thought maybe going over the whole files could help diagnose a failure in case, for instance, the only difference would be 2 lines that were swapped.